File: message/taggedmessage.go
package message

import "github.com/CotaPreco/Horus/tag"

type TaggedMessage struct {
	Message
	Tag tag.Tag
}

func NewTaggedMessage(tag tag.Tag, payload []byte) *TaggedMessage {
	return &TaggedMessage{
		Message{payload},
		tag,
	}
}

File: php/src/CotaPreco/Horus/Message/TaggedMessage.php
<?php

namespace CotaPreco\Horus\Message;

use CotaPreco\Horus\Tag\TagInterface;

/**
 * @author Andrey K. Vital <andreykvital@gmail.com>
 */
class TaggedMessage extends Message
{
    /**
     * @var TagInterface
     */
    private $tag;

    /**
     * @param TagInterface $tag
     * @param string       $message
     */
    public function __construct(TagInterface $tag, $message)
    {
        parent::__construct($message);

        $this->tag = $tag;
    }

    /**
     * @return TagInterface
     */
    public function getTag()
    {
        return $this->tag;
    }
}
